http.c

```diff
@@ -753,6 +753,12 @@ static CURL *get_curl_handle(void)
 	}
 #endif
 
+#if LIBCURL_VERSION_NUM >= 0x074400
+	if (!strcmp("schannel", http_ssl_backend) && http_schannel_check_revoke) {
```

Comment:
I think `http_ssl_backend` can be `NULL`... So we need to check for `if (http_ssl_backend && !strcmp...`

Also, I think it would be nicer to have the `#if LIBCURL_VERSION_NUM >= 0x074400` check inside the conditional block, so that we can add an `#else` clause where we print out a `warning("This cURL version (%s) does not support checkRevoke=false", curl_version);`

Finally, since we want to set *NO*_REVOKE, I think the `!http_schannel_check_revoke` wants to lose the `!`...

Reply:
Whoops, I meant the `http_schannel_check_revoke` in the `if` expression wants to have a `!` before...

But now I see that you did something similar in patch 3...

Reply:
Yeah, I might just rewrite patch 3 to avoid the worries about `NULL` here.